syne_tune/optimizer/schedulers/neuralbands/neuralband.py

```diff
@@ -0,0 +1,382 @@
+from syne_tune.optimizer.schedulers.hyperband_promotion import PromotionRungSystem
+from syne_tune.optimizer.schedulers.hyperband_rush import (
+    RUSHPromotionRungSystem,
+    RUSHStoppingRungSystem,
+)
+from syne_tune.optimizer.schedulers.hyperband_stopping import StoppingRungSystem
+from syne_tune.optimizer.schedulers.searchers.utils.default_arguments import (
+    check_and_merge_defaults,
+    Integer,
+    Boolean,
+    Categorical,
+    filter_by_key,
+    String,
+    Dictionary,
+    Float,
+)
+
+from syne_tune.optimizer.schedulers.searchers.searcher import BaseSearcher
+from syne_tune.optimizer.schedulers.searchers.searcher_factory import searcher_factory
+
+
+logger = logging.getLogger(__name__)
+
+def is_continue_decision(trial_decision: str) -> bool:
+    return trial_decision == SchedulerDecision.CONTINUE
+
+
+from syne_tune.optimizer.schedulers.hyperband import HyperbandBracketManager, _ARGUMENT_KEYS, _CONSTRAINTS
+from syne_tune.optimizer.scheduler import (
+    TrialScheduler,
+    SchedulerDecision,
+    TrialSuggestion,
+)
+
+from syne_tune.optimizer.schedulers.hyperband import _get_rung_levels, _is_positive_int, _sample_bracket
+
+from syne_tune.config_space import cast_config_values
+
+
+from syne_tune.optimizer.schedulers.searchers.bayesopt.datatypes.hp_ranges import (
+    HyperparameterRanges,
+)
+from syne_tune.optimizer.schedulers.searchers.bayesopt.datatypes.hp_ranges_factory import (
+     make_hyperparameter_ranges,
+)
+
+from syne_tune.optimizer.schedulers.neuralbands.networks import Exploitation
+from syne_tune.optimizer.schedulers.hyperband import HyperbandScheduler
+
+
+class NeuralbandScheduler(HyperbandScheduler):
+
+    def __init__(self, config_space, gamma = 0.01, nu = 0.01, step_size = 30, max_while_loop = 100,  **kwargs):
+        # Before we can call the superclass constructor, we need to set a few
+        # members (see also `_extend_search_options`).
+        # To do this properly, we first check values and impute defaults for
+        # `kwargs`.
+        
+        
+        super(NeuralbandScheduler, self).__init__(config_space, **kwargs)
+        self.kwargs = kwargs
+        
+        # encoding
+        self.hp_ranges = make_hyperparameter_ranges(config_space = self.config_space)
+        self.input_dim = self.hp_ranges.ndarray_size
+           
+        #neural network
+        self.net = Exploitation(dim = self.input_dim)
+        self.net.brackets = kwargs["brackets"]
+        
+        self.currnet_best_score = 1.0 
+        self.gamma = gamma
+        self.nu = nu
+        if self.mode == "min":
+            self.train_step_size = step_size
+        else:
+            self.train_step_size = 2
+                
+        # how many trails we train network once
+        self.max_while_loop = max_while_loop
+        
+ 
+    def _initialize_searcher_new(self): 
+        searcher = self.kwargs["searcher"]     
+        print("configurations run out and initialize the searcher", searcher)
```

Comment:
Maybe this should be come a logging output (e.g logging.debug(...))?

Reply:
Sure, I've delete this line.